core/src/main/java/org/apache/calcite/plan/RelOptCluster.java

```diff
@@ -177,6 +177,14 @@ public void setMetadataQuery(Supplier<RelMetadataQuery> mqSupplier) {
     return (M) mq;
   }
 
+  /**
+   * Returns the RelMetadataQuery supplier
+   * @return The supplier of RelMetadataQuery
+   */
+  public Supplier<RelMetadataQuery> getMetadataQuerySupplier() {
+    return this.mqSupplier;
+  }
+
```

Comment:
I am wondering if we should change `setMetadataQuery` to `setMetadataQuerySupplier` before this version is released out. Because  `setMetadataQuery` is not paired with `getMetadataQuery `, but paired with `getMetadataQuerySupplier`, which may confuse people.

Reply:
sounds good. will do.